client/my-sites/plans-features-main/components/plan-upsell-modal/paid-plan-is-required-dialog.tsx

```diff
@@ -1,74 +1,82 @@
 import { LoadingPlaceholder } from '@automattic/components';
+import { FREE_THEME } from '@automattic/design-picker';
 import { PlanButton } from '@automattic/plans-grid-next';
 import { useEffect, useState } from '@wordpress/element';
 import { useTranslate } from 'i18n-calypso';
 import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
-import {
-	ButtonContainer,
-	DialogContainer,
-	Heading,
-	SubHeading,
-	Row,
-	RowWithBorder,
-	DomainName,
-} from './components';
+import { getHidePlanPropsBasedOnThemeType } from '../utils/utils';
+import { ButtonContainer, DialogContainer, Heading, Row } from './components';
+import { PlanName, PlanDescription } from './components/plan-item';
 import SuggestedPlanSection from './components/suggested-plan-section';
 import { DomainPlanDialogProps, MODAL_VIEW_EVENT_NAME } from '.';
 
-export default function PaidPlanIsRequiredDialog( {
+export const PaidPlanIsRequiredDialog = ( {
 	paidDomainName,
 	generatedWPComSubdomain,
+	selectedThemeType,
 	onFreePlanSelected,
 	onPlanSelected,
-}: DomainPlanDialogProps ) {
+}: DomainPlanDialogProps ) => {
 	const translate = useTranslate();
 	const [ isBusy, setIsBusy ] = useState( false );
+	const isPaidTheme = selectedThemeType && selectedThemeType !== FREE_THEME;
+	const hidePlanProps = getHidePlanPropsBasedOnThemeType( selectedThemeType );
+
+	const getUpsellTitle = () => {
+		if ( isPaidTheme && paidDomainName ) {
```

Comment:
Not a blocker but maybe something to fix in a follow up, this `paidDomainName` isn't being set correctly. 

If you select a free domain and then a paid domain, it ignores the paid domain, if you select a paid domain and then a free domain it treats it like a paid domain.

![Screenshot 2024-12-18 at 16-55-25 Create a site — WordPress com](https://github.com/user-attachments/assets/09137ee0-b03f-47d7-a23b-453c0099bdb4)
![Screenshot 2024-12-18 at 16-55-17 Create a site — WordPress com](https://github.com/user-attachments/assets/1c11d6a9-ec47-4d9e-a1a0-e015756ef868)


It's set [here after a recent move](https://github.com/Automattic/wp-calypso/blob/ae438427d847bef507986eef7c28430ddeba0e47/client/landing/stepper/declarative-flow/internals/steps-repository/unified-plans/unified-plans-step.tsx#L494), but was [introduced originally over here](https://github.com/Automattic/wp-calypso/commit/a4e0338d25327ae3495f9015fce7f756f1a203fe#diff-f4523a33abda8414624dfd8061cffa9cc3b52e263f9a755706831b28b7b55cecR154) cc @aneeshd16 in case you can help figure out a fix.

Reply:
Interesting...it looks good on my side. I selected a free domain first and then a paid domain, but I can still see the following one:

<img width="658" alt="image" src="https://github.com/user-attachments/assets/11e949e3-c50e-4597-9340-3ca1e110c868" />

Reply:
If you have a premium theme if overrides and avoids this problem, if you only have a paid domain I think this triggers. Easier to test / trigger this in /start but /setup/onboarding has the same problem.

Reply:
Hi @lsl, I could not reproduce this issue. In both cases, it shows the same plans step. Here's a video:

**Free domain, then paid domain:**

https://github.com/user-attachments/assets/6ea04325-2e9e-4f03-8b49-c8d2d786bd73

**Paid domain, then free domain:**

https://github.com/user-attachments/assets/ba72882e-22dc-4af1-ba16-6495f37de524

Reply:
Here's what I'm seeing, are you able to replicate the same like this?

https://github.com/user-attachments/assets/a53d71ad-4a0d-449b-92c8-0648a40f1009

Reply:
I see. I could reproduce it by clicking on the Continue button before it finishes loading. If you wait for it to load, it shows the plans correctly. It seems like an issue with the domain mini cart? 🤔 I am not familiar with it, @paulopmt1 maybe?

https://github.com/user-attachments/assets/6414b8f7-6b6b-40b3-b29c-8b987a97e270

Reply:
Cool! I'll try to disable the button before it's loaded.

Reply:
Oh thanks for noticing that. :thinking: that might explain something else I was seeing with the select buttons too.

Reply:
Proposed fix here, https://github.com/Automattic/wp-calypso/pull/97681.